### PR TITLE
Renamed enum color to colors.

### PR DESCRIPTION
**Pull request by Remotion**
Added enum colors conversion to rgb struct.
Added colors_test.cpp.

Now it looks a bit strange. 
We have `enum color`  from format.h and `enum class colors` from colors.h.

### CODE REVIEW

**Comment by eliaskosunen** [comment]
`(v)print_colored` and `enum color` should probably be entirely replaced by this. They aren't even documented, so I don't think it would be a breaking change. This would also allow renaming the enum to the more logical singular-form `color`.

**Comment by Remotion** [comment]
Agree name `color` is better but then there are 2 different enums with same name. 
How can this be resolved then ? Rename or remove enum color from format.h  ?

**Comment by vitaut** [comment]
I suggest removing the old `color` and `print_colored`. Those are undocumented and experimental anyway and the new functionality seems strictly more general.

**Comment by Remotion** [comment]
What about  `fmt::printf(rgb, "");` ?  I could add this too but then colors.h will need to include printf.h.

**Comment by vitaut** [comment]
I don't think we need `fmt::printf(rgb, "")`. The whole point of `printf` is to simplify transition from legacy `printf` and the latter doesn't have color support.

**Comment by Remotion** [comment]
What do we miss here ?

File: include/fmt/colors.h [hunk 1]
```diff
@@ -70,18 +74,34 @@ FMT_FUNC void vprint_rgb(rgb fd, rgb bg, string_view format, format_args args) {
...
 
-// rgb foreground color
+/**
+  Formats a string and prints it to stdout using ANSI escape sequences to
+  specify foreground color 'fd'.
+  Example:
+    fmt::print(fmt::color::red, "Elapsed time: {0:.2f} seconds", 1.23);
+ */
 template <typename... Args>
 inline void print(rgb fd, string_view format_str, const Args & ... args) {
```
Comment (vitaut) [inline include/fmt/colors.h]: Why have both `print` and `print_rgb`?

**Comment by eliaskosunen** [comment]
You could try fixing the g++ 4.8 build. https://github.com/Remotion/fmt-fork/blob/74107c9bd8c1df2b74d3cb4c0d98f0dcc60813d7/include/fmt/colors.h#L39-L43

`FMT_CONSTEXPR` expands to `constexpr` when relaxed C++14 constexpr is available and `inline` otherwise. Because `constexpr` functions are implicitly `inline`, you should remove the extra `inline`.

File: include/fmt/colors.h (hunk 1 above)
Comment (Remotion) [reply include/fmt/colors.h]: print_rgb is now removed.

**Comment by vitaut** [comment]
Merged in https://github.com/fmtlib/fmt/commit/ce500635c7a0fbf899a2d158043d6fa8f04304b3 with minor tweaks to prevent breaking the API (we can remove the old color API during the next major release). Thanks!

**Comment by eliaskosunen** [comment]
Is it really necessary to wait for the next major release, though? The previous color API was not documented anywhere, so the feature wasn't really stabilized and it wouldn't really be a breaking change to remove it.

**Comment by vitaut** [comment]
It is still part of the public API, so I'd rather remove it in stages. Maybe not in the major version but at least we need to provide a good migration plan.

**Comment by vitaut** [comment]
I added a comment that the old color code is deprecated (https://github.com/fmtlib/fmt/blob/master/include/fmt/core.h#L1185) and we can make `FMT_EXTENDED_COLORS` the default in one of the minor versions.

**Comment by vitaut** [comment]
@Remotion, could you please review the updated [CONTIBUTING](https://github.com/fmtlib/fmt/blob/ebec00138bd10e949ff34955bcb096387cd32acc/CONTRIBUTING.md) document, particularly the part about licensing, and let me know if you agree with it being applied both to your contributions to {fmt}? The library is likely to be relicensed (#1073) so I'm collecting approval from all earlier contributors.

**Pull request closed without merging**